server/internal/server/vector_stores.go

```diff
@@ -0,0 +1,372 @@
+				Key:          k,
+				Value:        v,
+			}
+			if err := s.store.CreateCollectionMetadata(cm); err != nil {
+				return nil, status.Errorf(codes.Internal, "create collection metadata: %s", err)
+			}
+		} else if found.Value != v {
+			found.Value = v
+			if err := s.store.UpdateCollectionMetadata(found); err != nil {
+				return nil, status.Errorf(codes.Internal, "update collection metadata: %s", err)
+			}
+		}
+	}
+	for _, cm := range cms {
+		if _, ok := req.Metadata[cm.Key]; !ok {
+			if err := s.store.DeleteCollectionMetadata(cm.CollectionID, cm.Key); err != nil {
+				return nil, status.Errorf(codes.Internal, "delete collection metadata: %s", err)
+			}
+		}
+	}
+	cms, err = s.store.ListCollectionMetadataByCollectionID(c.CollectionID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "list collection metadata: %s", err)
+	}
+
+	// update collection
+	if req.Name != "" && req.Name != c.Name {
+		if err := s.vstoreClient.UpdateVectorStoreName(ctx, c.Name, req.Name); err != nil {
+			return nil, status.Errorf(codes.Internal, "update vector store: %s", err)
+		}
+		c.Name = req.Name
+	}
+	if ea := req.ExpiresAfter; ea != nil {
+		if err := validateExpiresAfter(ea); err != nil {
+			return nil, err
+		}
+		c.Anchor = store.ExpiresAfterAnchor(ea.Anchor)
+		c.ExpiresAfterDays = ea.Days
+	}
+	if err := s.store.UpdateCollection(c); err != nil {
+		return nil, status.Errorf(codes.Internal, "update collection: %s", err)
+	}
+
+	return toVectorStoreProto(c, cms), nil
+}
+
+// DeleteVectorStore deletes a vector store.
+func (s *S) DeleteVectorStore(
+	ctx context.Context,
+	req *v1.DeleteVectorStoreRequest,
+) (*v1.DeleteVectorStoreResponse, error) {
+	userInfo, err := s.extractUserInfoFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := getCollectionID(req.VectorStoreId)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := s.vstoreClient.DeleteVectorStore(ctx, req.VectorStoreId); err != nil {
+		if !errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, status.Errorf(codes.Internal, "delete collection: %s", err)
+		}
+		log.Printf("Collection %q not found in vector store server.", req.VectorStoreId)
+	}
+
+	// TODO(guangrui): Delete collection and collection metadata in a transaction.
+	if err := s.store.DeleteCollection(userInfo.ProjectID, id); err != nil {
+		return nil, status.Errorf(codes.Internal, "delete collection: %s", err)
+	}
+	if err := s.store.DeleteCollectionMetadataByCollectionID(id); err != nil {
+		return nil, status.Errorf(codes.Internal, "delete collection metadata: %s", err)
+	}
+	return &v1.DeleteVectorStoreResponse{
+		Id:      req.VectorStoreId,
+		Object:  vectorStoreObject,
+		Deleted: true,
+	}, nil
+}
+
+func toVectorStoreProto(c *store.Collection, cms []*store.CollectionMetadata) *v1.VectorStore {
+	m := map[string]string{}
+	for _, cm := range cms {
+		m[cm.Key] = cm.Value
+	}
+
+	return &v1.VectorStore{
+		Id:         fmt.Sprintf("%d", c.CollectionID),
```

Comment:
Slightly prefer having a vector ID separate from collection ID. We can then hide the internal implementation (and the ID name can follow the OpenAI API format).

```
{id: "vs_v9WodrpusSGehNzFAT1S1Gcp", object: "vector_store", name: null, status: "completed",…}
```

No need to change. I might make a change later if I started thinking more strongly.